### PR TITLE
In CreateTestingNS generate name upfront to avoid namespace leakage.

### LINKED ISSUE #71697: Framework.CreateNamespace can leak namespaces on retry
**What happened**:
In https://gubernator.k8s.io/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability-killer/83 the test failed due to namespace leakage:
1. First we created e2e-tests-load-30-nodepods-1-pfn9m, but never got response from api server (got 0-length response instead)
2. Then we retried and created e2e-tests-load-30-nodepods-1-m9tq5
3. We added e2e-tests-load-30-nodepods-1-m9tq5 to f.namespacesToDelete and leaked -pfn9m
4. The next test case failed because e2e-tests-load-30-nodepods-1-pfn9m wasn't deleted.

**What you expected to happen**:

While it's unclear why we got 0-length response instead (#71696 investigates that), we always can get error while the namespace was created (e.g. connection reset, timeout, io error).
We should handle this case in retrying correctly.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.14.0-alpha.0.793+dc9261bc3b6b81
- Cloud provider or hardware configuration: gce
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by mborsz**
**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Right now, when Namespaces().Create() api call fails with error like connection reset or similar, we have no idea whether the namespace was created or what is its name.
This can lead to namespace leakage and then failing the test.
With this PR we first chose name of the namespace and then try to create that namespace. In this way, retries will always use the same name.
**Which issue(s) this PR fixes**:

Fixes #71697

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -1052,6 +1052,25 @@ func WaitForPersistentVolumeClaimsPhase(phase v1.PersistentVolumeClaimPhase, c c
...
 
+// findAvailableNamespaceName random namespace name starting with baseName.
+func findAvailableNamespaceName(baseName string, c clientset.Interface) (string, error) {
+	var name string
+	err := wait.PollImmediate(Poll, 30*time.Second, func() (bool, error) {
+		name = fmt.Sprintf("%v-%v", baseName, randomSuffix())
+		_, err := c.CoreV1().Namespaces().Get(name, metav1.GetOptions{})
+		if err == nil {
+			// Already taken
+			return false, nil
```
Comment (wojtek-t) [inline test/e2e/framework/util.go]: I'm not following this. returning false, nil here, means we will retry again (and will probably get the same result). And in the end we will timeout.

I think you should return some kind of error here, which you will interpret differently outside.

Comment (mborsz) [reply test/e2e/framework/util.go]: Returning false here means that we will retry and with high probability pick another name (name is generated in each loop separately).

Comment (wojtek-t) [reply test/e2e/framework/util.go]: aah ok - we're generating name inside. Sorry. That makes sense now.

**Review by wojtek-t — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**